### PR TITLE
ENH: implement ``__skip_array_function__`` attribute for NEP-18

### PR DESCRIPTION
**Pull request by shoyer**
To clarify the implications of  https://github.com/numpy/numpy/pull/13305

### CODE REVIEW

**Comment by charris** [comment]
Can remove the draft status after NEP changes are accepted?

**Comment by shoyer** [comment]
> Can remove the draft status after NEP changes are accepted?

Yes, will do

**Comment by mattip** [comment]
@shoyer was the work to change  `public_api` to have clearer exceptions turned into a PR?

**Comment by charris** [comment]
@mattip I think I already merged that in  #13529. Or were you thinking of something else?

File: numpy/core/overrides.py [hunk 1]
```diff
@@ -152,6 +152,8 @@ def public_api(*args, **kwargs):
         if module is not None:
             public_api.__module__ = module
 
+        public_api.__skip_array_function__ = implementation
+
         return public_api
```
Comment (mattip) [inline numpy/core/overrides.py]: This will not merge cleanly after #13529

Comment (shoyer) [reply numpy/core/overrides.py]: fixed

**Comment by shoyer** [comment]
This is ready for review

**Comment by mattip** [comment]
LGTM

**Comment by shoyer** [comment]
Anyone else want to take a look? Otherwise I'll merge this tomorrow. I have a few small changes building on top of this that I want to get into NumPy 1.17.

**Pull request merged**